### PR TITLE
Use CMake FetchContent for Catch2

### PR DESCRIPTION
**Pull request by godexsoft**
Submodules are awkward and require the user of the library to clone the repo with special arg (`--recurse-submodules`) in order to fetch dependencies. CMake can do better than that if we use its FetchContent functionality. 

With the changes in this PR one can simply clone the repo and cmake as usual to get a working build without submodules.

### CODE REVIEW

**Comment by godexsoft** [comment]
Not too sure why it broke clang/release but I also don't have it setup on my machine to try debug it. Any ideas @Bronek ?

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -24,10 +24,7 @@ jobs:
         dnf -y install git cmake ninja-build clang gcc gcc-c++ libstdc++-static
         dnf -y clean all
 
-    # To enable submodules, git must be installed before actions/checkout@v3
     - uses: actions/checkout@v4
```
Comment (Bronek) [inline .github/workflows/build.yml:27]: can you move this step above the "Install prerequisites", please ? I had to put it in this non-standard position to get `git` installed first, but now we do not need this unusual order anymore.

Which reminds me - I guess we can also remove `git` installation in the other step.

**Review by Bronek — CHANGES_REQUESTED** [review]
nice, thank you - just one remark

**Comment by Bronek** [comment]
> Not too sure why it broke clang/release but I also don't have it setup on my machine to try debug it. Any ideas @Bronek ?

no idea, seems unrelated to this PR

File: .github/workflows/build.yml (hunk 1 above)
Comment (godexsoft) [reply .github/workflows/build.yml:27]: This does not work because FetchContent uses git under the hood.. i put it back now.

**Review by Bronek — APPROVED** [review]

**Pull request merged**